File: tasks_26/solutions/26-37.py

with open("26-k6.txt") as F:
  data = F.readlines()

N, K = map(int, data[0].split())

del data[0]
data = data[:N]

pairs = []
for i in range(N):
  pairs.append( tuple( map(int, data[i].split()) ) )

pairs.sort( key = lambda x: (x[1]/x[0], -x[1]) )

selected = pairs[:K]
print( sum( x[0] for x in selected ), end=" " )
weight = [x[0] for x in selected]
ind = weight.index( max(weight) )
print( selected[ind][1] )

File: tasks_26/solutions/26-37.pas
##
assign(input, '26-k6.txt');
var (N, K):= ReadInteger2;

var pairs := (1..N).Select(i->ReadString.ToIntegers)
                   .Skip(1).ToArray;
pairs.Sort( (x,y)-> x[1]/x[0]-y[1]/y[0] <> 0 ? 
                    sign(x[1]/x[0]-y[1]/y[0]) : y[1]-x[1] );

var selected := pairs[:K];

var weight := selected.Select( x->x[0] ).ToArray;
weight.Sum.Println;

var ind := weight.IndexOf( weight.Max );
selected[ind][1].Println;